src/DebugEngineHost.VSCode/VSCode/HandleCollection.cs

```diff
@@ -2,82 +2,68 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 
-using System;
-using System.Linq;
 using System.Collections.Generic;
 
 namespace Microsoft.DebugEngineHost.VSCode
 {
     public sealed class HandleCollection<T>
     {
-        private const int START_HANDLE = 1000;
+        private List<T> _collection = new List<T>();
 
-        private int _nextHandle;
-        private Dictionary<int, T> _handleMap;
-
-        public HandleCollection()
-        {
-            _nextHandle = START_HANDLE;
-            _handleMap = new Dictionary<int, T>();
-        }
-
-        public void Reset()
+        public void Clear()
         {
-            _nextHandle = START_HANDLE;
-            _handleMap.Clear();
+            _collection.Clear();
         }
 
         public int Create(T value)
         {
-            var handle = _nextHandle++;
-            _handleMap[handle] = value;
-            return handle;
+            _collection.Add(value);
+            return _collection.Count - 1;
         }
 
         public bool TryGet(int handle, out T value)
         {
-            if (_handleMap.TryGetValue(handle, out value))
+            if (handle >= _collection.Count)
             {
-                return true;
+                value = default;
+                return false;
             }
-            return false;
+
+            value = _collection[handle];
+            return true;
         }
 
         public bool TryGetFirst(out T value)
         {
             if (IsEmpty)
             {
-                value = default(T);
+                value = default;
                 return false;
             }
 
-            return TryGet(_handleMap.Keys.Min(), out value);
+            value = _collection[0];
+            return true;
         }
 
         public T this[int handle]
         {
             get
             {
-                T value;
-                if (!TryGet(handle, out value))
-                {
-                    throw new ArgumentOutOfRangeException("handle");
-                }
-
-                return value;
+                return _collection[handle];
             }
         }
 
         public bool Remove(int handle)
```

Comment:
Do we actually call this method? If so it isn't going to work with your new implementation.

Reply:
It's used in `HostMarshal.ReleaseCodeContextId`. You're right.